### DISCUSSION TITLE
Alias and Function config files

### DISCUSSION THREAD

**Discussion by trymeouteh** [nushell/nushell#13473] [Ideas]
I would like to suggest having Nu generate a `alias.nu` and `functions.nu` files when the user first run Nu and will decide where these scripts are stored.

These two scripts will be empty and can be used as a place for the user to add custom aliases and functions which can be accessed in the shell. Kinda like adding aliases and functions to `.bashrc` file for Bash but for Nu and having separate files to keep the custom aliases and functions in a separate file.

**Comment by ghost** [comment]
you can already do this manually: put `source alias.nu` in your `env.nu` and create the file yourself.

creating empty files by default seems a bit odd and since everybody has their own opinions as to how to structure the config (mine for example is 90% in json files) this sounds unnecessary controversal (why do we create a `functions.nu`, but no `external.nu`? etc).

**Reply by trymeouteh** [reply]
I would suggest at least adding the lines `source alias.nu` into the `env.nu` but having the `env.nu` file checking if the `alias.nu` file and `functions.nu` file exists so there no need to tamper with the `env.nu` file.

**Reply by ghost** [reply]
> so there no need to tamper with the env.nu file.

creating your own aliases is tampering. weather you tamper with your env or a custom one wont make a difference to a new user and more files are more confusing i think.

>  checking if the alias.nu file and functions.nu file exists

unless there has been a big change: if you have `source` or `use` anywhere in your script the file has to exist uppon SCRIPT LOAD not line execution. checking does exactly nothing.  
it would require checking in the `env.nu`, generating a temporary file (containing the `source` commands) and then sourcing that one in the `config.nu`.  
this would be way to workaround for an official example config in my opinion (its bad enough that `starship`, etc use that in their official implementations in my opinion)

**Reply by trymeouteh** [reply]
> creating your own aliases is tampering

By tampering, I mean modifying the `env.nu ` file. Kinda like modifying the `.bashrc` file to have it see if there is a `.bashalias` file and `.bashfunctions` file which is not the best since the system will update the file now and then.

And to have a official file for storing custom alias and an official file for storing functions would keep things neat and tidy.
